init/cli/lib.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"io/ioutil"
+	"os"
+	"syscall"
+
+	"github.com/palantir/pkg/cli"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+
+	"github.com/palantir/go-java-launcher/launchlib"
+)
+
+const (
+	outputFileFlag       = os.O_CREATE | os.O_WRONLY
+	truncOutputFileFlag  = outputFileFlag | os.O_TRUNC
+	appendOutputFileFlag = outputFileFlag | os.O_APPEND
+	outputFileMode       = 0666
+)
+
+var (
+	launcherStaticFile = "service/bin/launcher-static.yml"
+	launcherCustomFile = "var/conf/launcher-custom.yml"
+	pidfile            = "var/run/pids.yml"
+)
+
+type servicePids map[string]int
+
+type serviceStatus struct {
+	notRunningCmds map[string]launchlib.CmdWithContext
+	writtenPids    servicePids
+	runningProcs   map[string]*os.Process
+}
+
+func getServiceStatus(ctx cli.Context) (*serviceStatus, error) {
+	cmds, err := getConfiguredCommands(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get commands from static and custom configuration files")
+	}
+	writtenPids, runningProcs, err := getPidfileInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to determine running processes")
+	}
+	notRunningCmds := make(map[string]launchlib.CmdWithContext)
+	for name, cmd := range cmds {
+		if _, ok := runningProcs[name]; !ok {
+			notRunningCmds[name] = cmd
+		}
+	}
+	return &serviceStatus{notRunningCmds, writtenPids, runningProcs}, nil
+}
+
+func getPidfileInfo() (servicePids, map[string]*os.Process, error) {
+	pidfileBytes, err := ioutil.ReadFile(pidfile)
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
double negative - can we use `os.IsExist(err)`?

Reply:
and pull that out in a `pidfileExists` variable so that the else condition below reads `!pidfileExists` rather than calling the same / similar os function twice